test/framework/filetools.py

```diff
@@ -3560,6 +3607,126 @@ def test_create_unused_dir(self):
         self.assertEqual(path, os.path.join(self.test_prefix, 'file_0'))
         self.assertExists(path)
 
+    def test_create_unused_dirs(self):
+        """Test create_unused_dirs function."""
+        requested_paths = []
+        requested_paths.append(os.path.join(self.test_prefix, 'folder_a'))
+        requested_paths.append(os.path.join(self.test_prefix, 'folder_b'))
+        paths = ft.create_unused_dirs(requested_paths)
+        self.assertEqual(paths, requested_paths)
+        map(lambda p: self.assertExists(p), paths)
+
+        # Repeat with existing folder(s) should create new ones
+        for i in range(10):
+            requested_paths = []
+            requested_paths.append(os.path.join(self.test_prefix, 'folder_a'))
+            requested_paths.append(os.path.join(self.test_prefix, 'folder_b'))
+            paths = ft.create_unused_dirs(requested_paths)
+            self.assertEqual(paths, list(map(lambda p: p + '_%s' % i, requested_paths)))
```

Comment:
(Especially) with a lambda there usually isn't an advantage of using map instead of a list comprehension and the latter becomes likely better:
```suggestion
            self.assertEqual(paths, [f'{p}_{i}' for p in requested_paths])
```
E.g. it avoids pitfalls like not running the generator or trying to run it twice.   
I think we also usually use that in EB